Examples/Algorithms/TrackFinding/src/HoughTransformSeeder.cpp

```diff
@@ -0,0 +1,517 @@
+      continue;
+    }
+    if (!(m_cfg.sliceTester(meas->z, meas->layer, subregion)).value()) {
+      continue;
+    }
+
+    // This scans over y (pT) because that is more efficient in memory
+    for (unsigned y_ = 0; y_ < m_cfg.houghHistSize_y; y_++) {
+      unsigned y_bin_min = y_;
+      unsigned y_bin_max = (y_ + 1);
+
+      // Find the min/max x bins
+      auto xBins =
+          yToXBins(y_bin_min, y_bin_max, meas->radius, meas->phi, meas->layer);
+      // Update the houghHist
+      for (unsigned y = y_bin_min; y < y_bin_max; y++) {
+        for (unsigned x = xBins.first; x < xBins.second; x++) {
+          houghHist(y, x).first++;
+          houghHist(y, x).second.insert(index);
+        }
+      }
+    }
+  }
+
+  return houghHist;
+}
+
+ActsExamples::HoughHist ActsExamples::HoughTransformSeeder::createHoughHist(
+    int subregion) const {
+  ActsExamples::HoughHist houghHist(m_cfg.houghHistSize_y,
+                                    m_cfg.houghHistSize_x);
+
+  for (unsigned i = 0; i < m_cfg.nLayers; i++) {
+    HoughHist layerHoughHist = createLayerHoughHist(i, subregion);
+    for (unsigned x = 0; x < m_cfg.houghHistSize_x; ++x) {
+      for (unsigned y = 0; y < m_cfg.houghHistSize_y; ++y) {
+        if (layerHoughHist(y, x).first > 0) {
+          houghHist(y, x).first++;
+          houghHist(y, x).second.insert(layerHoughHist(y, x).second.begin(),
+                                        layerHoughHist(y, x).second.end());
+        }
+      }
+    }
+  }
+
+  return houghHist;
+}
+
+bool ActsExamples::HoughTransformSeeder::passThreshold(
+    HoughHist const& houghHist, unsigned x, unsigned y) const {
+  // Pass window threshold
+  unsigned width = m_cfg.threshold.size() / 2;
+  if (x < width || (houghHist.size(1) - x) < width) {
+    return false;
+  }
+  for (unsigned i = 0; i < m_cfg.threshold.size(); i++) {
+    if (houghHist(y, x - width + i).first < m_cfg.threshold[i]) {
+      return false;
+    }
+  }
+
+  // Pass local-maximum check, if used
+  if (m_cfg.localMaxWindowSize != 0) {
+    for (int j = -m_cfg.localMaxWindowSize; j <= m_cfg.localMaxWindowSize;
+         j++) {
+      for (int i = -m_cfg.localMaxWindowSize; i <= m_cfg.localMaxWindowSize;
+           i++) {
+        if (i == 0 && j == 0) {
+          continue;
+        }
+        if (y + j < houghHist.size(0) && x + i < houghHist.size(1)) {
+          if (houghHist(y + j, x + i).first > houghHist(y, x).first) {
+            return false;
+          }
+          if (houghHist(y + j, x + i).first == houghHist(y, x).first) {
+            if (houghHist(y + j, x + i).second.size() >
+                houghHist(y, x).second.size()) {
+              return false;
+            }
+            if (houghHist(y + j, x + i).second.size() ==
+                    houghHist(y, x).second.size() &&
+                j <= 0 && i <= 0) {
+              return false;  // favor bottom-left (low phi, low neg q/pt)
+            }
+          }
+        }
```

Comment:
Do you think combining the according `if`s in `else if`s could make this section more readable?

```suggestion
        if (y + j < houghHist.size(0) && x + i < houghHist.size(1)) {
          if (houghHist(y + j, x + i).first > houghHist(y, x).first) {
            return false;
          } else if (houghHist(y + j, x + i).first == houghHist(y, x).first) {
            if (houghHist(y + j, x + i).second.size() >
                houghHist(y, x).second.size()) {
              return false;
            } else if (houghHist(y + j, x + i).second.size() ==
                       houghHist(y, x).second.size()) {
              if (j <= 0 && i <= 0) {
                return false;  // favor bottom-left (low phi, low neg q/pt)
              }
            }
          }
        }
```

Reply:
Let's leave it like this, for the moment.